### PR TITLE
Bugfix/njv 331 change name on all devices 450

### PR DESCRIPTION
**Pull request by paykoob**
resolve https://github.com/actorapp/actor-platform/issues/450

### CODE REVIEW

File: actor-server/actor-core/src/main/scala/im/actor/server/user/UserCommandHandlers.scala [hunk 1]
```diff
@@ -158,8 +158,7 @@ private[user] trait UserCommandHandlers {
         val update = UpdateUserNameChanged(userId, name)
         for {
           relatedUserIds ← getRelations(userId)
-          _ ← seqUpdatesExt.broadcastSingleUpdate(relatedUserIds, update)
-          seqstate ← seqUpdatesExt.deliverSingleUpdate(user.id, update)
+          (seqstate, _) ← seqUpdatesExt.broadcastOwnSingleUpdate(userId, relatedUserIds, update)
```
Comment (rockjam) [inline actor-server/actor-core/src/main/scala/im/actor/server/user/UserCommandHandlers.scala]: @paykoob what is the difference here?

**Comment by rockjam** [comment]
@paykoob Thank you for contribution! Will merge you pull request tomorrow. Left you a question in server side code

File: actor-server/actor-core/src/main/scala/im/actor/server/user/UserCommandHandlers.scala (hunk 1 above)
Comment (rockjam) [inline actor-server/actor-core/src/main/scala/im/actor/server/user/UserCommandHandlers.scala]: Oh, I see. It does the same but in one line. nice 👍

**Comment by ex3ndr** [comment]
Reject. Made incompatible changes in core, restored unnecessary binding in profile fragment.

**Comment by ex3ndr** [comment]
Ah i see, something is wrong with user updating from difference. It doesn't overwrite new user object from reponse.

**Comment by ex3ndr** [comment]
Probably server-side issue, but not sure.

**Comment by paykoob** [comment]
@ex3ndr  I asked a question on https://github.com/actorapp/actor-platform/issues/450 for reason of ignoring the 'NameChanged'.

this is problem as I undrastand:
when both bob and alice are online and bob changes his name, server send update to alice and she can see the new name and every things works well.

but if alice is offline, after being online she try to get difference. server send her all of user updates, including NameChanged as same as AvatarChanged.

but in core (GetDifferentCombiner) NameChanged ignores and AvatarChanged counts. and onAvatarChanged has been called but not onNameChanged!

I cant see an issue in server side as much as i could!

**Comment by paykoob** [comment]
and for binding about this is UX manner. we can just keep it in our team.

**Comment by ex3ndr** [comment]
> but in core (GetDifferentCombiner) NameChanged ignores and AvatarChanged counts. and onAvatarChanged has been called but not onNameChanged!

This should be ignored because we already applying updates from difference. May be something broken when we started to strip entities from difference

**Comment by ex3ndr** [comment]
Considering binding you need to show/hide about/nicknames when they are not present. You will eventually get empty strings in your profile screen.

**Comment by paykoob** [comment]
> Considering binding you need to show/hide about/nicknames when they are not present. You will eventually get empty strings in your profile screen.

I was aware about that. and binding performs if they are present. (and after setting the about/nickname you cant set them empty)

**Comment by paykoob** [comment]
> This should be ignored because we already applying updates from difference. May be something broken when we started to strip entities from difference

yep, this is stripping issue. 
but in this situation you should ignore AvatarChanged and any of user/group desc changes too.

if it should be handled on server-side, we should consider this updates (Avatar and name ...) in SequenceServiceImpl:

```
 private def getUsersGroups(
    userIds:       Set[Int],
    groupIds:      Set[Int],
    optimizations: Seq[ApiUpdateOptimization.Value]
  )(implicit client: AuthorizedClientData): Future[(Vector[ApiUser], Vector[ApiGroup])] = {
    if (optimizations.contains(ApiUpdateOptimization.STRIP_ENTITIES))
      Future.successful((Vector.empty, Vector.empty))
    else
      for {
        groups ← Future.sequence(groupIds.toVector map (GroupExtension(actorSystem).getApiStruct(_, client.userId)))
        // TODO: #perf optimize collection operations
        allUserIds = userIds ++ groups.foldLeft(Vector.empty[Int]) { (ids, g) ⇒ ids ++ g.members.flatMap(m ⇒ Vector(m.userId, m.inviterUserId)) :+ g.creatorUserId }
        users ← Future.sequence(allUserIds.toVector map (UserUtils.safeGetUser(_, client.userId, client.authId))) map (_.flatten)
      } yield (users, groups)
  } 
```

and IMHO this is not pretty solution. or we should remove stripping optimization.

**Comment by ex3ndr** [comment]
I think your solution if OK and i will accept your changes when you will fix (or remove) binding.

**Comment by paykoob** [comment]
showing empty string didn't occur in this PR (because we checked they shouldn't be null) , but I improved the PR and added a commit.

**Comment by ex3ndr** [comment]
Merged, but one last case is not handled - when person remove about or username. If it is not possible in android app than there is a bug in android.

**Pull request merged**